src/hotspot/share/gc/g1/heapRegion.hpp

```diff
@@ -183,6 +186,8 @@ class HeapRegion : public CHeapObj<mtGC> {
 
   // Returns the object size for all valid block starts
   // and the amount of unallocated words if called on top()
+  template<bool RESOLVE>
+  size_t block_size(const HeapWord* p) const;
```

Comment:
There is non-templated `block_size` here too, just below. Is the non-templated version used? If so, do you want to drop it and say `template<bool RESOLVE = true>` instead?